weavertest/init.go

```diff
@@ -65,6 +71,18 @@ var (
 // AllRunners returns a slice of all builtin weavertest runners.
 func AllRunners() []Runner { return []Runner{Local, RPC, Multi} }
 
+// FakeComponent records the implementation to use for a specific component type.
+type FakeComponent struct {
+	intf reflect.Type
+	impl any
+}
+
+// Fake arranges to use impl as the implementation for the component type T.
+// The result is typically placed in Runner.Fakes.
+func Fake[T any](impl T) FakeComponent {
```

Comment:
Because `Fake` infers `T` from the type of `impl`, if you forget to explicitly pass `T`, I think `Fake` can sometimes infer the wrong type. Taking `simple_test.go` as an example, if the user were to write the following:

```go
fake := &fakeDest{}
runner.Fakes = append(runner.Fakes, weavertest.Fake(fake))
```

`Fake` infers `T` to be `*fakeDest`. What do you think about changing `Fake` to the following to force the user to explicitly pass `T`?

```go
func Fake[T any](impl any) FakeComponent {
    if _, ok := impl.(T); !ok {
        panic(...)
    }
    return FakeComponent{intf: reflect.TypeOf((*T)(nil)).Elem(), impl: impl}
}
```

A user can still easily mess things up though, and it complicates the signature, so it might not be worth it. Up to you.

Reply:
I like requiring T, so I will switch to your suggestion and ping back when that is done.